### PR TITLE
feature: add azure analysis service scanner

### PR DESCRIPTION
**Pull request by vanwinkelseppe**
# Description

Add a Analytics Service scanner with the following rules:

- Diagnostic settings
- SLA
- Tags
- CAF Naming conventions

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #220 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [X] Code compiles correctly
* [X] Created/updated tests
* [X] Unit tests passing
* [X] Linting
* [] Ran documentation script

### CODE REVIEW

**Comment by vanwinkelseppe** [comment]
have to update the rule documentation but waiting till end to reduce amount of commits to update for every PR/rebase

File: README.md [hunk 1]
```diff
@@ -63,6 +63,7 @@ To learn more about the recommendations used by **Azure Quick Review (azqr)**, y
 
 **Azure Quick Review (azqr)** currently supports the following Azure services:
 
+* Azure Analytics Service
```
Comment (cmendible) [inline README.md:66]: Azure Analysis Services

File: cmd/azqr/as.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+	"github.com/spf13/cobra"
+)
+
+func init() {
+	scanCmd.AddCommand(asCmd)
+}
+
+var asCmd = &cobra.Command{
+	Use:   "as",
+	Short: "Scan Azure Analytics Service",
```
Comment (cmendible) [inline cmd/azqr/as.go:18]:
```suggestion
	Short: "Scan Azure Analysis Services",
```

File: cmd/azqr/as.go [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+	Long:  "Scan Azure Analytics Service",
```
Comment (cmendible) [inline cmd/azqr/as.go:19]:
```suggestion
	Long: "Scan Azure Analysis Services",
```

File: docs/content/en/docs/Overview/_index.md [hunk 4]
```diff
@@ -58,6 +58,7 @@ To learn more about the recommendations used by **Azure Quick Review (azqr)**, y
 
 **Azure Quick Review (azqr)** currently supports the following Azure services:
 
+* Azure Analytics Service
```
Comment (cmendible) [inline docs/content/en/docs/Overview/_index.md:61]:
```suggestion
* Azure Analysis Services
```

**Review by cmendible — CHANGES_REQUESTED** [review]

File: internal/scanners/as/rules.go [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+					sla = "99.9%"
+				}
+				return sla == "None", sla
+			},
+			Url: "https://www.microsoft.com/licensing/docs/view/Service-Level-Agreements-SLA-for-Online-Services",
+		},
+		"as-003": {
+			Id:             "as-003",
+			Category:       scanners.RulesCategoryHighAvailability,
+			Recommendation: "Azure Analysis SKU",
```
Comment (cmendible) [inline internal/scanners/as/rules.go:47]:
```suggestion
			Recommendation: "Azure Analysis Service SKU",
```

File: internal/scanners/as/rules.go [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+			Impact:         scanners.ImpactHigh,
+			Eval: func(target interface{}, scanContext *scanners.ScanContext) (bool, string) {
+				i := target.(*armanalysisservices.Server)
+				return false, string(*i.SKU.Name)
+			},
+			Url: "https://azure.microsoft.com/en-us/pricing/details/databricks/",
```
Comment (cmendible) [inline internal/scanners/as/rules.go:53]: Please check the url

**Review by cmendible — CHANGES_REQUESTED** [review]
Thanks for this one! We are almost there!

**Comment by vanwinkelseppe** [comment]
> Thanks for this one! We are almost there!

@cmendible all done, waiting to run docs until it's looking good

File: internal/scanners/as/as.go [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+// Init - Initializes the AnalysisServicesScanner
+func (c *AnalysisServicesScanner) Init(config *scanners.ScannerConfig) error {
+	c.config = config
+	var err error
+	c.client, err = armanalysisservices.NewServersClient(config.SubscriptionID, config.Cred, config.ClientOptions)
+	return err
+}
+
+// Scan - Scans all Analytics Services in a Resource Group
```
Comment (cmendible) [inline internal/scanners/as/as.go:25]:
```suggestion
// Scan - Scans all Analysis Services in a Resource Group
```

**Review by cmendible — CHANGES_REQUESTED** [review]

**Review by cmendible — APPROVED** [review]
LGTM

**Review by cmendible — CHANGES_REQUESTED** [review]
new scanner must be added to the `GetScanners()` func.

**Review by cmendible — APPROVED** [review]
LGTM

**Pull request merged**